test/buffer/worker/utils_test.go

```diff
@@ -111,9 +111,13 @@ func (ts *testSuite) IterateMetrics(fn func(<-chan *dto.MetricFamily)) {
 // there can be another line between them, but the order must be preserved.
 func (ts *testSuite) WaitForLogMessages(timeout time.Duration, lines string) bool {
 	expected := `%A` + strings.ReplaceAll(strings.TrimSpace(lines), "\n", "\n%A") + `%A`
-	return assert.Eventually(ts.t, func() bool {
+	ok := assert.Eventually(ts.t, func() bool {
 		return wildcards.Compare(expected, ts.logsOut.String()) == nil
-	}, timeout, 100*time.Millisecond, ts.logsOut.String())
+	}, timeout, 100*time.Millisecond, "see all log messaged bellow")
+	if !ok {
+		ts.t.Logf("all log messages that were searched for:\n%s", ts.logsOut.String())
+	}
+	return ok
 }
```

Comment:
Drobny fix v testoch.

Do `assert.Eventually(` je mozne volitelne pridat nejaku vysvetlujucu `msg`, ktora sa vypise v teste ak assert spadne.

Vid zmazane  `ts.logsOut.String())`.

---------------

Cielom bolo vypisat vsertky logs, ak sa dana expected msg nenajde.

---------

Chyba je v tom, ze  `ts.logsOut.String())` sa vyhodnocuje predtym, ako sa spusti callback / skonci assert.
Takze je to k nicomu.

----------

Musi sa to vypisat oddelene, ak assert skoncil a spadol.